src/ORM/Association.php

```diff
@@ -815,6 +815,11 @@ protected function _formatAssociationResults($query, $surrogate, $options)
         $formatters = $surrogate->formatResults();
 
         if (!$formatters || empty($options['propertyPath'])) {
+            if ($formatters) {
```

Comment:
Yeah it's a bit weird. The alternatives seem odd too, like if empty property path return, if no formatters return else apply formatters and return. Or if formatters and empty property path apply formatters then return else return.

Seems like there'd be an else in there... Or a whole extra outside double conditional.